tests/tests_zone.yml

```diff
@@ -0,0 +1,125 @@
+        when: not firewall_lib_result.changed
+
+      # ENSURE STATE AGAIN
+
+      - name: Setup firewalld again
+        include_role:
+          name: linux-system-roles.firewall
+        vars:
+          firewall_setup_default_solution: no
+          firewall:
+            - { zone: 'internal',
+                service: [ 'tftp', 'ftp' ],
+                port: [ '443/tcp', '443/udp' ],
+                trust: [ 'interface0', 'interface1' ],
+                masq: [ 'interface2', 'interface3' ],
+                forward_port: [ '447/tcp;;1.2.3.4',
+                                '448/tcp;;1.2.3.5' ],
+                state: 'enabled' }
+
+      - name: Fail on newly changes
+        fail: msg="FAILED"
+        when: firewall_lib_result.changed
+
+      # VERIFY
+
+      - name: Verify firewalld zone internal services
+        command: firewall-cmd --permanent --zone=internal --list-services
+        register: result
+        failed_when: result.failed
+                     or "tftp" not in result.stdout
+                     or "ftp" not in result.stdout
+
+      - name: Verify firewalld zone internal ports
+        command: firewall-cmd --permanent --zone=internal --list-ports
+        register: result
+        failed_when: result.failed
+                     or "443/tcp" not in result.stdout
+                     or "443/udp" not in result.stdout
+
+      - name: Verify firewalld zone internal forward ports
+        command: firewall-cmd --permanent --zone=internal --list-forward-ports
+        register: result
+        failed_when: result.failed
+                     or "port=447:proto=tcp:toport=:toaddr=1.2.3.4"
+                        not in result.stdout
+                     or "port=448:proto=tcp:toport=:toaddr=1.2.3.5"
+                        not in result.stdout
+
+      - name: Verify firewalld zone external interfaces
+        command: firewall-cmd --permanent --zone=external --list-interfaces
+        register: result
+        failed_when: result.failed
+                     or "interface2" not in result.stdout
+                     or "interface3" not in result.stdout
+
+      - name: Verify firewalld zone trusted interfaces
+        command: firewall-cmd --permanent --zone=trusted --list-interfaces
+        register: result
+        failed_when: result.failed
+                     or "interface0" not in result.stdout
+                     or "interface1" not in result.stdout
+
+    always:
+
+      # CLEANUP: RESET TO ZONE DEFAULTS
+
+      - name: Reset to zone defaults
+        shell:
+          cmd: |
+            firewall-cmd --permanent --load-zone-defaults=internal
+            firewall-cmd --permanent --load-zone-defaults=external
+            firewall-cmd --permanent --load-zone-defaults=trusted
+            firewall-cmd --reload
+
+    when: ansible_distribution == "Fedora" or
+      ((ansible_distribution == "CentOS" or ansible_distribution == "RedHat")
+       and ansible_distribution_major_version >= "7")
```

Comment:
`and ansible_distribution_major_version|int >= 7` - to use numeric comparison instead of string (but string is ok for this particular case)

Reply:
FIxed